util/dofile.pl

```diff
@@ -38,12 +38,27 @@
 
 # -o ORIGINATOR
 #		declares ORIGINATOR as the originating script.
-getopt('o', \%opts);
+# -s FILE       Source the file and eval it (for variables, usually)
+getopt('osi', \%opts);
 
-my @autowarntext = ("WARNING: do not edit!",
-                    "Generated"
-                    . (defined($opts{o}) ? " by ".$opts{o} : "")
-                    . (scalar(@ARGV) > 0 ? " from ".join(", ",@ARGV) : ""));
+my @autowarntext = (
+    "WARNING: do not edit!",
+    "Generated"
+        . (defined($opts{o}) ? " by $opts{o}" : "")
+        . (defined($opts{s}) ? " with " . $opts{s} : "")
+        . (scalar(@ARGV) > 0 ? " from " .join(", ", @ARGV) : "")
+);
+
+if (defined($opts{s})) {
+    local $/ = undef;
+    open VARS, $opts{s} or die "Couldn't open $opts{s}, $!";
+    my $contents = <VARS>;
+    close VARS;
+    eval $contents;
+    die $@ if $@;
+}
+die "Must have input files"
+   if defined($opts{i}) and scalar(@ARGV) == 0;
```

Comment:
This is actually valid just for `@ARGV` being empty, no need to check if `-i` has been given.

Reply:
I don't understand this comment.

Reply:
What I'm trying to say is that this is sufficient:

``` perl
die "Must have input files"
   if scalar(@ARGV) == 0;
```

Reply:
You sure?  It does default to stdin if no files are given (see line 67)
I am quite happy to do that if you want it :)

Reply:
I don't think that we are ever using it without at least one input file. Maybe I was thinking differently at some point, so hold on with this for a time while I check (won't be immediately, sorry)

Reply:
Ok, a bit of back pedalling here.  Please ignore this concern, it's always nice to have a stdin possibility.